gsdplugin.c

```diff
@@ -841,16 +841,18 @@ static int read_gsd_timestep(void *mydata, int natoms, molfile_timestep_t *ts)
             }
         else
             {
-            ts->A = box[0]; ts->B = box[1]; ts->C = box[2];
+            // define lattice constants in terms of box size and tilt factors
+            const double xy = (double)box[3];
+            const double xz = (double)box[4];
+            const double yz = (double)box[5];
+            const double norm1 = sqrt(1.0 + xy*xy);
+            const double norm2 = sqrt(1.0 + xz*xz + yz*yz);
+            
+            ts->A = box[0]; ts->B = (float)norm1*box[1]; ts->C = (float)norm2*box[2];
```

Comment:
To make sure orthorhombic boxes still get their sizes set exactly, please put this code back in the `if` branch below and include these new lines there:
```
ts->A = box[0];
ts->B = (float)(norm1*box[1]);
ts->C = (float)(norm2*box[2]);
```
Then, also add the original line to the `else` branch where the angles are also set exactly:
```
ts->A = box[0]; ts->B = box[1]; ts->C = box[2];
```